test/convection/model/test_validation.rb

```diff
@@ -0,0 +1,184 @@
+require 'test_helper'
+require 'json'
+require 'pp'
+
+class TestValidations < Minitest::Test
+
+  def setup
+    @within_limits = ::Convection.template do
```

Comment:
`setup` probably isn't where you want to create all these templates. The method is run before every test method (methods named `test_*`). Move each of these declarations into their respective test.